frontend/vite-project/src/pages/ProductDetails.tsx

```diff
@@ -32,6 +33,20 @@ const ProductDetails: React.FC = () => {
     fetchPlant();
   }, [id]);
 
+  useEffect(() => {
+    if (!id) return;
+    const fetchTasks = async () => {
+      try {
+        const res = await axios.get(`http://localhost:5000/plants/${id}/tasks`);
+        setTasks(res.data);
+      } catch (err) {
+        setTasks([]);
+        console.error("Error fetching tasks", err);
+      }
+    };
+    fetchTasks();
+  }, [id]);
+
```

Comment:
There is probably a better way to do this, but 

```suggestion
  const months = ["", "January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
```